node/risk-app/server/src/js/map-page/map-page.js

```diff
@@ -0,0 +1,251 @@
+import { scenarioDisplayUpdate, handleScroll, handleArrowClick } from './scenario-bars.js'
+import { openKey, closeKey, showOrHideAdvancedToggleText, toggleAdvancedOptions, handleRadioChange, selectedOption } from './map-controls.js'
+import { adjustPosition } from './screen-size-adjustments.js'
+import { mapPageConsts } from './constants.js'
+
+/* global mapCategories */
+class MapController {
+  constructor (categories) {
+    this._categories = categories
+  }
+
+  /**
+ * setCurrent
+ * @param {string} ref The ref of either a category or map. If a category ref is passed, the first map in that category is used.
+ */
+  setCurrent (ref) {
+  // Work out the current category and map
+    let category, map, defaultCategory, defaultMap
+    for (let i = 0; i < this._categories.length; i++) {
+      category = this._categories[i]
+      if (i === 0) {
+        defaultCategory = category
+      }
+
+      if (category.ref === ref) {
+        this.currMap = category.maps[0]
+        this.currCategory = category
+        return
+      }
+
+      for (let j = 0; j < category.maps.length; j++) {
+        map = category.maps[j]
+        if (i === 0 && j === 0) {
+          defaultMap = map
+        }
+
+        if (map.ref === ref) {
+          this.currMap = map
+          this.currCategory = category
+          return
+        }
+      }
+    }
+
+    this.currMap = defaultMap
+    this.currCategory = defaultCategory
+  }
+}
+
+function mapPage () {
+  getInitialKeyOptions()
+
+  function getParameterByName (name) {
+    name = name.replace(/[[]/, '\\[').replace(/\]/, '\\]')
+    const regex = new RegExp(`[\\?&]${name}=([^&#]*)`)
+    const results = regex.exec(window.location.search)
+    return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '))
+  }
+  const measurements = document.querySelectorAll('.govuk-radios__inputs')
+  const map = document.getElementById('map')
+  const body = document.body
+
+  const easting = parseInt(getParameterByName('easting'), 10)
+  const northing = parseInt(getParameterByName('northing'), 10)
+  const hasLocation = !!easting
+
+  mapPageConsts.maps.loadMap((hasLocation && [easting, northing]))
+
+  // This function updates the map to the radio button you select (extent, depth, velocity)
+  function setCurrent (ref) {
+    const mapController = new MapController(mapCategories.categories)
```

Comment:
If you include `window.mapCategories.categories` here, that will make explicit where the data is coming from, stop the SonarCloud warning, and remove the need for the global declaration on Line 6.